File: Data Structures & Algorithms/binary-tree-preorder-traversal/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:

    vector<int> preorderTraversal(TreeNode* root) {
        vector<int>res;
        preOrder(root,res);
        return res;
    }
private:
void preOrder(TreeNode* node,vector<int>&res){
    if(!node) return;
    res.push_back(node->val);
    if(node->left)preOrder(node->left,res);
    if(node->right)preOrder(node->right,res);

}
};

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        self.res=-1
        self.cnt=0
        def dfs(root,x):
            if root is None or self.cnt==x:
                return
            dfs(root.left,x)
            self.cnt+=1
            if self.cnt==x:
                self.res=root.val
            dfs(root.right,x)
        dfs(root,k)
        return self.res

        